src/richie/plugins/lti_consumer/models.py

```diff
@@ -0,0 +1,119 @@
+"""
+LTI consumer plugin models
+"""
+from urllib.parse import unquote
+
+from django.conf import settings
+from django.db import models
+from django.utils.translation import gettext_lazy as _
+
+import exrex
+from cms.models.pluginmodel import CMSPlugin
+from cms.utils import get_current_site
+from oauthlib import oauth1
+
+from .defaults import INSTRUCTOR, STUDENT
+
+
+class LtiConsumer(CMSPlugin):
+    """
+    LTI consumer plugin model.
+    """
+
+    url = models.URLField(_("LTI url"), null=True, blank=True)
+    context_id = models.CharField(max_length=100, null=True, blank=True)
```

Comment:
Actually, on second thought, you don't need this field since we always want to send the active site... so you should get it while building the LTI request and just send it...

Reply:
But in case of multiple site, we couldn't retrieve lti content across all sites, no ?

Reply:
why not? anyway I'm not sure if a given page/placeholder/plugin can be part of multiple sites... but the idea is to avoid storing in db something that is already known (and could change by the way)...

Reply:
That's the point : if the current site domain can change and so the context_id, I suppose that LTI content couldn't be retrieved…
Storing context_id ensures that the one used for adding lti content will always be used in the student view.

Reply:
I would say that in this case, we should have a setting with a list of predefined contexts to which an LTI consumer plugin can be associated :man_shrugging:  I find it weird to get the site info and save it in a field without any formatting. This is opening the way to context hell?